src/Campaigns/ServiceProvider.php

```diff
@@ -123,6 +124,9 @@ private function registerActions(): void
         Hooks::addAction('givewp_campaign_page_created', AssociateCampaignPageWithCampaign::class);
         Hooks::addAction('give_form_duplicated', Actions\AssignDuplicatedFormToCampaign::class, '__invoke', 10, 2);
 
+        Hooks::addAction('givewp_donation_form_deleting', PreventDeleteDefaultForm::class);
```

Comment:
This action passes a donation model instance, looks like your trying to access an ID.

Also, I don't think the legacy tables use the form model do they? What if we just hid the trash link instead of being heavy handed?

Reply:
@jonwaldstein  You are right! This mistake happened because I was testing different hooks and did the push using the wrong one. The `before_delete_post action` works for both list views. Resolved in 07cc80e8c191723831d568ffe2e738f16e1ecee6

Reply:
@jonwaldstein The advantage of this approach, instead of hiding the link, is that it also blocks accidental exclusions through bulk action for both list views.